src/main/java/mate/academy/spring/boot/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,77 @@
+package mate.academy.spring.boot.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Positive;
+import lombok.RequiredArgsConstructor;
+import mate.academy.spring.boot.dto.cartitem.CartItemDto;
+import mate.academy.spring.boot.dto.cartitem.CartItemQuantityRequestDto;
+import mate.academy.spring.boot.dto.cartitem.CreateCartItemRequestDto;
+import mate.academy.spring.boot.dto.shoppingcart.ShoppingCartDto;
+import mate.academy.spring.boot.model.User;
+import mate.academy.spring.boot.service.CartItemService;
+import mate.academy.spring.boot.service.ShoppingCartService;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping Cart manager", description = "Endpoints for managing user's shopping cart")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/cart")
+@Validated
+public class ShoppingCartController {
+    private final CartItemService cartItemService;
+    private final ShoppingCartService shopCartService;
+
+    @PreAuthorize("hasRole('USER')")
+    @PostMapping
+    @Operation(summary = "Add a some book to the shopping cart",
+            description = "Create a new cartItem entity in the database")
+    public CartItemDto createCartItem(@RequestBody @Valid CreateCartItemRequestDto requestDto,
```

Comment:
```suggestion
    public ShoppingCartDto createCartItem(@RequestBody @Valid CreateCartItemRequestDto requestDto,
```

Reply:
not fixed.
Please fix all Elena's comments